src/api/routes/auth.routes.ts

```diff
@@ -0,0 +1,319 @@
+import { Request, Response, NextFunction, Router } from 'express';
+import passport from 'passport';
+import { Strategy as LocalStrategy, IVerifyOptions } from 'passport-local';
+import bcrypt from 'bcrypt';
+import zxcvbn from 'zxcvbn';
+import validator from 'validator';
+import { SALT_ROUNDS } from '../middlewares.js';
+import AppDataSource from '../data-source.js';
+import { User } from '../entities/user.js';
+
+// Define interfaces for request payloads
+interface RegistrationPayload {
+  username?: string;
+  email?: string;
+  password?: string;
+}
+
+const router = Router();
+
+// NEW: Local map for registration rate limiting, specific to this file
+const successfulRegistrationsLocal = new Map<string, number>();
+
+passport.use(
+  new LocalStrategy(
+    async (
+      username,
+      password,
+      done: (
+        error: unknown,
+        user?: Express.User | false, // Express.User should pick up the global definition
```

Comment:
This is where you use the globally-defined variable. What if you made it locally defined?